rules/opentitan.bzl

```diff
@@ -183,44 +293,28 @@ elf_to_scrambled = rule(
 def opentitan_binary(
         name,
         platform = OPENTITAN_PLATFORM,
-        per_device_deps = {
-            "verilator": ["//sw/device/lib/arch:sim_verilator"],
-            "dv": ["//sw/device/lib/arch:sim_dv"],
-            "fpga_nexysvideo": ["//sw/device/lib/arch:fpga_nexysvideo"],
-            "cw310": ["//sw/device/lib/arch:fpga_cw310"],
-        },
-        signing_keys = {
-            "test_key_0": "//sw/device/silicon_creator/mask_rom/keys:test_private_key_0",
-        },
         output_bin = True,
         output_disassembly = True,
-        output_signed = False,
-        output_scrambled = False,
         **kwargs):
     """A helper macro for generating OpenTitan binary artifacts.
+
     This macro is mostly a wrapper around cc_binary, but creates artifacts
-    for each of the keys in `per_device_deps`.  The actual artifacts
-    created are an ELF file, a BIN file, the disassembly and the scrambled
-    ROM image.  Each of these output targets performs a bazel transition to
-    the RV32I toolchain to build the target under the correct compiler.
+    compatible with OpenTitan binaries. The actual artifacts created are an ELF
+    file, a BIN file, and the disassembly. Each of these output targets performs
+    a bazel transition to the RV32I toolchain to build the target under the
+    correct compiler.
     Args:
       @param name: The name of this rule.
       @param platform: The target platform for the artifacts.
-      @param per_device_deps: The deps for each of the execution environments.
       @param output_bin: Whether or not to emit a BIN file.
       @param output_disassembly: Whether or not to emit a disassembly file.
-      @param output_signed: Whether or not to emit a signed binary.
-      @param output_scrambled: Whether or not to emit a SCR file.
       @param **kwargs: Arguments to forward to `cc_binary`.
     Emits rules:
-      For each device in per_device_deps entry:
-        cc_binary     named: name_device
-        obj_transform named: name_device_elf
-        optionally:
-          obj_transform       named: name_device_bin
-          elf_to_dissassembly named: name_device_dis
-          elf_to_scrambled    named: name_device_scr
-      filegroup named: name
+      cc_binary             named: <name>
+      optionally:
+        obj_transform       named: <name>_bin
+        elf_to_dissassembly named: <name>_dis
```

Comment:
Something I'd like us to do at some point is to, instead (or along with) this output, to also emit the output of running `clang -s` for various files, which can be much richer than just disassembling the linked binary... not something for this CL, probably...

Reply:
That makes sense. I agree, this PR is already approaching the 500-line mark. Maybe file a separate issue for that?

Reply:
Yes, separate issue.  Also: the binary rules should be modified to emit a map file.  This may be something we'll need to bring up with the bazel folks, as the cc_binary rule doesn't seem to preserve the mapfile as an output if you ask it to generate one with `-Wl,-Map=foo.map`.